packages/acceptance-tests/pkg-tests-specs/sources/commands/workspaces/foreach.test.js

```diff
@@ -135,7 +135,7 @@ describe(`Commands`, () => {
 
           try {
             await run(`install`);
-            ({code, stdout, stderr} = await run(`workspaces`, `foreach`, `--parallel`, `--topological`, `node`, `-p`, `require("./package.json").name`, {cwd: `${path}/packages/workspace-d`}));
+            ({code, stdout, stderr} = await run(`workspaces`, `foreach`, `--parallel`, `--topological`, `node`, `-p`, `require("./package.json").name`, {cwd: `${path}/packages/workspace-c`}));
           } catch (error) {
             ({code, stdout, stderr} = error);
```

Comment:
A lot of the tests in this file follow this rather odd pattern where they will catch and assert against errors when they are intending to test the non-error path.

That doesn't seem desirable -- it seems like they should assume success (if they aren't a test intended to test error behavior) so that (1) erroneously-updated snapshots like this don't happen in again the future, (2) it's clear which branch the test is actually intended to assert against and (3) the test fails hard with an exception when something breaks.

I'm happy to update all the tests in this file accordingly if you agree.

Reply:
Indeed, that seems to be an oversight 🤔 cc @larixer 

> I'm happy to update all the tests in this file accordingly if you agree.

Would be very welcome, thanks!

Reply:
We should be able to simplify it even more by passing the promise directly to `expect`
```js
await expect(run(`[...]`)).resolves.toMatchSnapshot();
```

Reply:
Wasn't sure if there was a stylistic reason that these fields were being destructured, so I erred on the side of leaving them. Changed.